src/P3_size_distribution.jl

```diff
@@ -1,361 +1,339 @@
+    return log(a) - b_μ_1 * log_λ + SF.loggamma(b_μ_1) + log(abs(q_x - q_y))
 end
 
 """
-    L_over_N_gamma(p3, F_rim, F_liq, log_λ, th)
-
- - p3 - a struct with P3 scheme parameters
- - F_rim - rime mass fraction (L_rim / L_ice) [-]
- - F_liq - liquid fraction (L_liq / L_p3_tot) [-]:
-    - zero if solving for ice core shape parameters
- - log_λ - logarithm of the slope parameter of N′ gamma distribution
- - μ - shape parameter of N′ gamma distribution
- - th - P3 scheme thresholds() output tuple (D_cr, D_gr, ρ_g, ρ_d)
-
-Returns L/N for all values of D (sum over all regimes).
-Eq. 5 in Morrison and Milbrandt (2015).
+    get_μ(dist)
+    get_μ(state, log_λ)
+    get_μ(params, log_λ)
+    
+Compute the slope parameter μ
+
+# Arguments
+- `dist::P3Distribution`: [`P3Distribution`](@ref) object
+- `state::P3State`: [`P3State`](@ref) object
+- `params::PSP3`: [`CMP.ParametersP3`](@ref) object
+- `log_λ`: The log of the slope parameter [log(1/m)]
 """
-function L_over_N_gamma(
-    p3::PSP3,
-    F_rim::FT,
-    F_liq::FT,
-    log_λ::FT,
-    μ::FT,
-    th = (; D_cr = FT(0), D_gr = FT(0), ρ_g = FT(0), ρ_d = FT(0)),
-) where {FT}
+get_μ(dist::P3Distribution) = get_μ(get_state(dist), dist.log_λ)
+get_μ(state::P3State, log_λ) = get_μ(get_parameters(state), log_λ)
+get_μ(params::PSP3, log_λ) = get_μ(params.slope, log_λ)
 
-    D_th = D_th_helper(p3)
-    λ = exp(log_λ)
-    N = CO.Γ(1 + μ) / (λ^(1 + μ))
-    return ifelse(
-        F_rim == FT(0),
-        (p3_F_liq_average(
-            F_liq,
-            L_s(p3, p3.ρ_i, μ, λ, FT(0), D_th) + L_rz(p3, μ, λ, D_th),
-            L_liq(p3, μ, λ),
-        )) / N,
-        (p3_F_liq_average(
-            F_liq,
-            L_s(p3, p3.ρ_i, μ, λ, FT(0), D_th) +
-            L_n(p3, μ, λ, D_th, th.D_gr) +
-            L_s(p3, th.ρ_g, μ, λ, th.D_gr, th.D_cr) +
-            L_r(p3, F_rim, μ, λ, th.D_cr),
-            L_liq(p3, μ, λ),
-        )) / N,
-    )
-end
+get_μ((; a, b, c, μ_max)::SlopePowerLaw, log_λ) =
+    clamp(a * exp(log_λ)^b - c, 0, μ_max)
+get_μ((; μ)::SlopeConstant, _) = μ
 
 """
-    DSD_μ_approx(p3, L, N, ρ_r, F_rim, F_liq)
-
- - p3 - a struct with P3 scheme parameters
- - L - ice content [kg/m3]
- - N - total ice number concentration [1/m3]
- - ρ_r - rime density (L_rim/B_rim) [kg/m^3]
- - F_rim - rime mass fraction (L_rim / L_ice) [-]
- - F_liq - liquid fraction (L_liq / L_p3_tot) [-]:
-    - zero if solving for ice core shape parameters
-Returns the approximated shape parameter μ for a given q and N value
-"""
-function DSD_μ_approx(
-    p3::PSP3,
-    L::FT,
-    N::FT,
-    ρ_r::FT,
-    F_rim::FT,
-    F_liq::FT,
-) where {FT}
-    # Get thresholds for given F_rim, ρ_r
-    th = thresholds(p3, ρ_r, F_rim)
+    log_LdN₀(state, log_λ)
 
-    # Get min and max lambda values
-    λ_0 = μ_to_λ(p3, FT(0))
-    λ_6 = μ_to_λ(p3, p3.μ_max)
+Compute `log(L/N₀)` given the `state` and `log_λ`
+"""
+function log_LdN₀(state::P3State{FT}, log_λ) where {FT}
```

Comment:
I'm wondering if `log_L_over_N₀` or ``log_L_div_N₀` would be better. `dN` could be confusing because you are using `dD1` for integrating over things in other places